### PR TITLE
Add json-item types

### PR DESCRIPTION
**Pull request by riyafa**

### CODE REVIEW

File: vxquery-core/src/main/java/org/apache/vxquery/functions/builtin-operators.xml [hunk 1]
```diff
@@ -855,15 +855,15 @@
 
     <!-- opext:array-constructor($expression as item()*) as array() -->
     <operator name="opext:array-constructor">
-        <param name="expression" type="item()*"/>
-        <return type="item()"/>
+        <param name="expression" type="json-item()*"/>
```
Comment (prestoncarman) [inline vxquery-core/src/main/java/org/apache/vxquery/functions/builtin-operators.xml]: param -- item()*

File: vxquery-core/src/main/java/org/apache/vxquery/functions/builtin-operators.xml [hunk 2]
```diff
@@ -855,15 +855,15 @@
...
+        <return type="json-item()"/>
```
Comment (prestoncarman) [inline vxquery-core/src/main/java/org/apache/vxquery/functions/builtin-operators.xml]: return -- array()

File: vxquery-core/src/main/java/org/apache/vxquery/functions/builtin-operators.xml [hunk 3]
```diff
@@ -855,15 +855,15 @@
...
         <runtime type="scalar" class="org.apache.vxquery.runtime.functions.jsonitem.ArrayConstructorScalarEvaluatorFactory"/>
     </operator>
 
     <!-- opext:object-constructor($expression as item()*) as object() -->
     <operator name="opext:object-constructor">
-        <param name="expression" type="item()*"/>
-        <return type="item()"/>
+        <param name="expression" type="json-item()*"/>
+        <return type="json-item()"/>
```
Comment (prestoncarman) [inline vxquery-core/src/main/java/org/apache/vxquery/functions/builtin-operators.xml]: return -- object()

File: vxquery-core/src/main/java/org/apache/vxquery/functions/builtin-operators.xml [hunk 4]
```diff
@@ -855,15 +855,15 @@
...
         <runtime type="scalar" class="org.apache.vxquery.runtime.functions.jsonitem.ArrayConstructorScalarEvaluatorFactory"/>
     </operator>
 
     <!-- opext:object-constructor($expression as item()*) as object() -->
     <operator name="opext:object-constructor">
-        <param name="expression" type="item()*"/>
-        <return type="item()"/>
+        <param name="expression" type="json-item()*"/>
```
Comment (prestoncarman) [inline vxquery-core/src/main/java/org/apache/vxquery/functions/builtin-operators.xml]: item()*

File: vxquery-core/src/main/java/org/apache/vxquery/types/TypeUtils.java [hunk 5]
```diff
@@ -27,39 +27,45 @@ public static SequenceType createSequenceType(String str) {
     }
 
     public static SequenceType createSequenceType(StaticContext sCtx, String str) {
-        str = str.trim();
+        String s = str.trim();
```
Comment (westmann) [inline vxquery-core/src/main/java/org/apache/vxquery/types/TypeUtils.java]: No need to change this now, but I would have preferred, if you would have used "str" for the new variable and changed the name of the parameter as that would have made it clearer what actually has changed.

File: vxquery-core/src/main/java/org/apache/vxquery/functions/builtin-operators.xml [hunk 6]
```diff
@@ -856,14 +856,14 @@
     <!-- opext:array-constructor($expression as item()*) as array() -->
     <operator name="opext:array-constructor">
         <param name="expression" type="item()*"/>
-        <return type="item()"/>
+        <return type="object()"/>
```
Comment (westmann) [inline vxquery-core/src/main/java/org/apache/vxquery/functions/builtin-operators.xml]: Should this be "array()"?

File: vxquery-core/src/main/java/org/apache/vxquery/types/TypeUtils.java (hunk 5 above)
Comment (riyafa) [inline vxquery-core/src/main/java/org/apache/vxquery/types/TypeUtils.java]: This change was initially done by Preston--I have now made the new variable str

Comment (prestoncarman) [inline vxquery-core/src/main/java/org/apache/vxquery/types/TypeUtils.java]: Sorry Riyafa, I got you in trouble. :-)

**Comment by prestoncarman** [comment]
Please squash these changes and make a nice commit message.

**Comment by westmann** [comment]
Looks good to me as well (after the squash :) ).

**Comment by prestoncarman** [comment]
I merged the squashed version of this change. Please close.

**Pull request closed without merging**